python/lsst/ts/dsm/dsm_csc.py

```diff
@@ -136,20 +135,21 @@ async def handle_summary_state(self):
                 self.cleanup_simulation()
 
             self.telemetry_loop_task.cancel()
-            if not self.telemetry_watcher.closed:
-                self.telemetry_watcher.unwatch(self.telemetry_directory)
-                self.telemetry_watcher.close()
+            if self.telemetry_watch is not None:
+                self.telemetry_notifier.rm_watch(self.telemetry_watch)
+                self.telemetry_watch = None
+                self.telemetry_notifier.close()
```

Comment:
If I am reading your code correctly, you close the notifier whenever you leave ENABLED state, but only create the notifier when you start the CSC. This is a surprising lack of symmetry, and my brief perusal of the asyncinotify docs suggests it's a bug: that once you close a notifier you cannot use it again.

If you should only close the notifier when you quit this CSC then please move this `close` call to the `close_tasks` method.

If this code is correct then please add a code comment here explaining what is going on, as I think other readers of the code will have the same question.

Reply:
Yes, this was a bug. I've fixed it according to your suggestion.